src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -79,17 +85,25 @@ public AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryI
             : PartitionAdditionalFiles(allFiles, analysisConfig);
     }
 
-    private List<FileInfo> GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
+    private FileInfo[] GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
         directoryWrapper
-            .EnumerateFiles(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
+            .EnumerateDirectories(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
+            .Concat([projectBaseDir.FullName]) // also include the root directory
+            .Where(x => !IsExcludedDirectory(x))
```

Comment:
Use the `EnumerateDirectories` that returns `DirectoryInfo`: https://learn.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.enumeratedirectories?view=net-8.0
That way, you do not have to split in `IsExcludedDirectory`.